src/patterns/ec2-app.test.ts

```diff
@@ -117,12 +117,54 @@ describe("the GuEC2App pattern", function () {
       access: { scope: AccessScope.PUBLIC },
       certificateProps: getCertificateProps(),
       monitoringConfiguration: {
-        tolerated5xxPercentage: 5,
         snsTopicName: "test-topic",
+        http5xxAlarm: {
+          tolerated5xxPercentage: 5,
+        },
+        unhealthyInstancesAlarm: false,
+      },
+      userData: "",
+    });
+    //The shape of this alarm is tested at construct level
+    expect(stack).toHaveResourceOfTypeAndLogicalId("AWS::CloudWatch::Alarm", /^High5xxPercentageAlarm.+/);
+  });
+
+  it("creates an UnhealthyInstancesAlarm if the user enables it", function () {
+    const stack = simpleGuStackForTesting();
+    const app = "test-gu-ec2-app";
+    new GuEc2App(stack, {
+      applicationPort: GuApplicationPorts.Node,
+      app,
+      access: { scope: AccessScope.PUBLIC },
+      certificateProps: getCertificateProps(),
+      monitoringConfiguration: {
+        snsTopicName: "test-topic",
+        http5xxAlarm: false,
+        unhealthyInstancesAlarm: true,
+      },
+      userData: "",
+    });
+    //The shape of this alarm is tested at construct level
+    expect(stack).toHaveResourceOfTypeAndLogicalId("AWS::CloudWatch::Alarm", /^UnhealthyInstancesAlarm.+/);
+  });
+
+  it("Skips alarm creation if the user explicitly opts-out", function () {
+    const stack = simpleGuStackForTesting();
+    const app = "test-gu-ec2-app";
+    new GuEc2App(stack, {
+      applicationPort: GuApplicationPorts.Node,
+      app,
+      access: { scope: AccessScope.PUBLIC },
+      certificateProps: getCertificateProps(),
+      monitoringConfiguration: {
+        snsTopicName: "test-topic",
```

Comment:
I would expect real users to use `{ noMonitoring: true }` in this scenario (as the `snsTopicName` is pointless), but this allows us to validate the behaviour in a single test.